.drone.star

```diff
@@ -1711,6 +1668,112 @@ def redisService():
 		},
 	}]
 
+def cacheOcisPipeline(ctx):
+	return[{
+		'kind': 'pipeline',
+		'type': 'docker',
+		'name': 'cache-ocis',
+		'workspace' : {
+			'base': '/var/www/owncloud',
+			'path': config['app']
+		},
+		'steps':
+			buildOCISCache() +
+			cacheOcis() +
+			listRemoteCache(),
+		'depends_on': [],
+		'trigger': {
+			'ref': [
+				'refs/pull/**',
```

Comment:
Seems like I forgot to add it, I'll fix it

Reply:
But now that I think about it, we only need to run this on pull requests. Once the cache is built on the pull request we don't need to do that again on the merge commits.

Reply:
As far as I know objects in the cache Bucket (`cache_s3_endpoint`) will be pruned after 14 days. As a consequence CI on master would stop working after 14 days (if there is no oCIS commid id bump)!? This is why we always need to check if the cache is already populated.

Reply:
`cache_s3_endpoint` is referencing the minio server https://cache.owncloud.com not the bucket. Everything inside the cache bucket will be pruned every 1d (`mc find local/cache --older-than 1d --exec "mc rm {}`). Only logs have a longer lifetime of 14 d (`mc find local/logging --older-than 14d --exec "mc rm {}`). In this case we don't use the cache bucket as screenshots need to be public available and should have a longer lifetime as they are linked in PR's.

Reply:
Fixed it, Now that we check the cache and build the ocis image based on that, it makes sense to run this on master as well. That should work also even if dont bump ocis commit id for 2 weeks. Also it doesn't really hurts to run this on master because the actual building of the binary doesnt happens if its already in cache.